test/e2e/apps/job.go

```diff
@@ -817,6 +818,58 @@ var _ = SIGDescribe("Job", func() {
 		gomega.Expect(jobs.Items).To(gomega.BeEmpty(), "Found job %v", jobName)
 	})
 
+	ginkgo.It("should check the ready of a pods", func(ctx context.Context) {
+		backoff := 0
+		ginkgo.By("Creating a job")
+		job := e2ejob.NewTestJob("notTerminate", "all-ready", v1.RestartPolicyNever, parallelism, completions, nil, int32(backoff))
+
+		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
+		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Ensuring job has ready pods")
+		err = e2ejob.WaitForJobReady(ctx, f.ClientSet, f.Namespace.Name, job.Name, ready)
+		framework.ExpectNoError(err, "failed to ensure job ready in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Ensuring pods for job exist")
+		pods, err := e2ejob.GetJobPods(ctx, f.ClientSet, f.Namespace.Name, job.Name)
+		framework.ExpectNoError(err, "failed to get pod list for job in namespace: %s", f.Namespace.Name)
+		podsReady := int32(0)
+		for _, pod := range pods.Items {
+			for _, c := range pod.Status.Conditions {
+				if c.Type == v1.PodReady && c.Status == v1.ConditionTrue {
+					podsReady++
+				}
+			}
+		}
+		gomega.Expect(podsReady).To(gomega.Equal(parallelism), "expected %d running job pods, but got %d", parallelism, podsReady)
+	})
+
+	ginkgo.It("should check the ready of a pods for failed job", func(ctx context.Context) {
```

Comment:
I'm hesitant we need this test, maybe it's fine to drop it entirely. We have already integration & unit tests for this feature. 

So, I think a sanity test that the field is set to parallelism should be enough. If we want to see in test different values for the field, maybe  expand the previous scenario with the following: 
1. create the job suspended, verify .status.ready=0
2. unsuspend the job, verify .status.ready=parallelism (as currently)
3. suspend the job again and verify .status.ready=0